components/link/Link.js

```diff
@@ -0,0 +1,45 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import s from './theme.css';
+
+let LinkComponent;
+
+class Link extends PureComponent {
+  static propTypes = {
+    children: PropTypes.node.isRequired,
+    className: PropTypes.string,
+  };
+
+  static defaultProps = {
+    className: '',
+  };
+
+  static use (LinkComponentToUse) {
```

Comment:
This `Link.use(...)` implementation makes it impossible to have different types of `Link` components within the same component. For example: a React router link, and a normal link in the same component. I think this should be possible.

Reply:
We also came to that conclusion yesterday. I just tested the following, so maybe that would work with importing Link components too;

```js
// fileA.js
export const foo = "some value";

// fileB.js
import { foo as bar, foo as baz } from './fileA';
console.log(bar); // "some value"
console.log(baz); // "some value"
```

Reply:
Your test actually proofs nothing and is quite useless :D (it just proofs that you can import the same thing twice under a different name, but it will still be the same think in the end, I guess it's just a reference or something).

I just tried it out, and it doesn't work. Here's some pseudo code 💅 

```js
import { Heading1, Heading2 } from 'HeadingShizzle';
import { Link as Link1, Link as Link2 } from 'LinkShizzle';

Link1.use(Heading1);
Link2.use(Heading2);

...

<Link1 /> // will use a Heading2 (while we expected a Heading1)
<Link2 /> // will use a Heading2 (which is what we expected)
```

Reply:
A factory is probably the solution?